File: C++/Cristian Maldonado/Matemática/A_MOD_X_EQUAL_B.cpp
#include <iostream>
#include <math.h>

using namespace std;

void modEquation(int a, int b){
    if(a < b){
        cout<<"No hay solucion"<<endl;
        return;
    }
    if(a == b){
        cout<<"Hay infinitas soluciones"<<endl;
    }
    int cont = 0;
    int n = a - b;
    int y = sqrt(a - b);
    for(int i = 1; i <= y; i++){
        if(n % i == 0){
            if(n / i > b){
                cont++;
                cout<<n/i<<endl;
            }
            if(i > b){
                cont++;
                cout<<i<<endl;
            }
        }
    }
    if(y * y == n && y > b){
        cont--;
    }
    cout<<cont<<endl;
}

int main()
{
    int a = 32, b = 2;
    modEquation(a, b);
    return 0;
}

File: Codigos estudiantes por lenguaje/Java/Cristian Maldonado/Matemática/A_MOD_X_EQUAL_B.java
public class A_MOD_X_EQUAL_B {

    static void modEquation(int a, int b) {
        if (a < b) {
            System.out.println("No hay solucion");
            return;
        }
        if (a == b) {
            System.out.println("Hay infinitas soluciones");
        }
        int count = 0;
        int n = a - b;
        int y = (int) Math.sqrt(a - b);
        for (int i = 1; i <= y; i++) {
            if(n % i == 0){
                if(n / i > b){
                    count++;
                    System.out.println(n / i);
                }
                if(i > b){
                    count++;
                    System.out.println(i);
                }
            }
        }
        if(y * y == n && y > b){
            count--;
        }
        System.out.println(count);
    }

    public static void main(String[] args) {
        int a = 32, b = 2;
        //21 % x = 5;
        modEquation(a, b);
    }
}
